docs/architecture/adr-027-ibc-wasm.md

```diff
@@ -100,7 +100,7 @@ packaged inside a payload which is json serialized and passed to `callContract`
 array of bytes returned by the smart contract. This data is deserialized and passed as return argument.
 
 ```go
-func (c *ClientState) CheckProposedHeaderAndUpdateState(context sdk.Context, marshaler codec.BinaryMarshaler, store sdk.KVStore, header exported.Header) (exported.ClientState, exported.ConsensusState, error) {
+func (c *ClientState) CheckProposedHeaderAndUpdateState(context sdk.Context, marshaler codec.BinaryMarshaler, store sdk.KVStore, header exported.ClientMessage) (exported.ClientState, exported.ConsensusState, error) {
```

Comment:
Shouldn't this be split up into two functions or we will allow the ADR authors to update?